src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -283,7 +286,18 @@ public function getSearchContext(GridField $gridField)
                     . " or implement a getDefaultSearchContext() method on $modelClass"
                 );
             }
-            $this->searchContext = $singleton->getDefaultSearchContext();
+
+            $list = $gridField->getList();
+            $searchContext = $singleton->getDefaultSearchContext();
+
+            // In case we are working with an ArrayList we need to convert the search context into a basic search context
+            // This is because the scaffolded filters are intended to use ORM for data searching,
+            // they rely on DataList functionality which is not available on an ArrayList
```

Comment:
```suggestion
            // In case we are working with a list not backed by the database we need to convert the search context into a BasicSearchContext
            // This is because the scaffolded filters use the ORM for data searching
```